lightning/src/offers/invoice.rs

```diff
@@ -461,6 +461,7 @@ for InvoiceBuilder<'a, DerivedSigningPubkey> {
 #[derive(Clone)]
 pub struct UnsignedBolt12Invoice {
 	bytes: Vec<u8>,
+	experimental_bytes: Vec<u8>,
```

Comment:
We end up writing `experimental_bytes` into `bytes` when signing / converting to `Bolt12Invoice`, so that would require finding the right place in `bytes` to insert the signature and shuffling the experimental bytes down. Alternatively, we'd allocate a new `Vec` to use for `Bolt12Invoice` and write to it as needed.